From c955e478df5a4131f96605021512de0267281ec5 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Tue, 3 Nov 2020 23:54:00 -0500
Subject: [PATCH] src/google/protobuf/port_def: use _WIN32 for
 PROTOBUF_MAYBE_CONSTEXPR

mingw-w64's std::mutex does not seem to have constexpr applied
---
 src/google/protobuf/port_def.inc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/google/protobuf/port_def.inc b/src/google/protobuf/port_def.inc
index 8b39d6989..807f19799 100644
--- a/src/google/protobuf/port_def.inc
+++ b/src/google/protobuf/port_def.inc
@@ -460,7 +460,7 @@
 #endif  // _MSC_VER
 
 #if defined(__clang__) || defined(__GNUC__) || defined(_MSC_VER)
-// Don't let Objective-C Macros interfere with proto identifiers with the same 
+// Don't let Objective-C Macros interfere with proto identifiers with the same
 // name.
 #pragma push_macro("DEBUG")
 #undef DEBUG
@@ -563,7 +563,7 @@
 // Some constructors can't be constexpr under MSVC, but given that MSVC will not
 // do constant initialization of globals anyway we can omit `constexpr` from
 // them. These constructors are marked with PROTOBUF_MAYBE_CONSTEXPR
-#if defined(_MSC_VER)
+#if defined(_WIN32)
 #define PROTOBUF_MAYBE_CONSTEXPR
 #else
 #define PROTOBUF_MAYBE_CONSTEXPR constexpr
-- 
2.28.0

